clang-tools-extra/clang-tidy/modernize/MinMaxUseInitializerListCheck.cpp

```diff
@@ -0,0 +1,274 @@
+            : nullptr;
+
+    if (InitList) {
+      Result.Args.append(InitList->inits().begin(), InitList->inits().end());
+      Result.First = *ArgIterator;
+      Result.Last = *ArgIterator;
+
+      // check if there is a comparison argument
+      std::advance(ArgIterator, 1);
+      if (ArgIterator != Call->arguments().end())
+        Result.Compare = *ArgIterator;
+
+      return Result;
+    }
+  } else {
+    // if it has 3 arguments then the last will be the comparison
+    Result.Compare = *(std::next(Call->arguments().begin(), 2));
+  }
+
+  if (Result.Compare)
+    Result.Args = SmallVector<const Expr *>(llvm::drop_end(Call->arguments()));
+  else
+    Result.Args = SmallVector<const Expr *>(Call->arguments());
+
+  Result.First = Result.Args.front();
+  Result.Last = Result.Args.back();
+
+  return Result;
+}
+
+static SmallVector<FixItHint>
+generateReplacements(const MatchFinder::MatchResult &Match,
+                     const CallExpr *TopCall, const FindArgsResult &Result,
+                     const bool IgnoreNonTrivialTypes,
+                     const unsigned long IgnoreTrivialTypesOfSizeAbove) {
+  SmallVector<FixItHint> FixItHints;
+
+  const QualType ResultType = TopCall->getDirectCallee()
+                                  ->getReturnType()
+                                  .getNonReferenceType()
+                                  .getUnqualifiedType()
+                                  .getCanonicalType();
+
+  // check if the type is trivial
+  const bool isResultTypeTrivial = Match.Context->getBaseElementType(ResultType)
+                                       .isTrivialType(*Match.Context);
+  const SourceManager &SourceMngr = *Match.SourceManager;
+  const LangOptions &LanguageOpts = Match.Context->getLangOpts();
+
+  if ((!isResultTypeTrivial && IgnoreNonTrivialTypes))
+    return FixItHints;
+
+  if (isResultTypeTrivial &&
+      // size in bits divided by 8 to get bytes
+      Match.Context->getTypeSize(ResultType) / 8 >
+          IgnoreTrivialTypesOfSizeAbove)
+    return FixItHints;
+
+  for (const Expr *Arg : Result.Args) {
+    const auto *InnerCall = dyn_cast<CallExpr>(Arg->IgnoreParenImpCasts());
+
+    // If the argument is not a nested call
+    if (!InnerCall) {
+      // check if typecast is required
+      const QualType ArgType = Arg->IgnoreParenImpCasts()
+                                   ->getType()
+                                   .getUnqualifiedType()
+                                   .getCanonicalType();
+
+      if (ArgType == ResultType)
+        continue;
+
+      const StringRef ArgText = Lexer::getSourceText(
+          CharSourceRange::getTokenRange(Arg->getSourceRange()), SourceMngr,
+          LanguageOpts);
+
+      Twine Replacement = llvm::Twine("static_cast<")
+                              .concat(ResultType.getAsString(LanguageOpts))
+                              .concat(">(")
+                              .concat(ArgText)
+                              .concat(")");
```

Comment:
Thanks for the heads up! After taking a look at other checks, it seems like it would be okay to skip assigning `ResultType.getAsString(LanguageOpts)` to a local variable. Storing it locally might  also create some reference lifetime issues.

Reply:
It doesn't mean that other checks are correct. Twine saves only pointer to std::string, it doesn't make a copy. Therefore in this case after call to last concat result of getAsString will be destroyed, if you would add call to .str() at the end of this chain it would be ok, but currently this is just a bug, additionally most probably because return type is small, it fit into stack memory.

Reply:
And this is reported by address sanitizer:
```
==335149==ERROR: AddressSanitizer: stack-use-after-scope on address 0x725ae7653890 at pc 0x56da3506075a bp 0x7ffccd4af960 sp 0x7ffccd4af958
READ of size 8 at 0x725ae7653890 thread T0
    #0 0x56da35060759 in print /fpwork/llvm-project/llvm/lib/Support/Twine.cpp:165:21
    #1 0x56da35060759 in llvm::Twine::printOneChild(llvm::raw_ostream&, llvm::Twine::Child, llvm::Twine::NodeKind) const /fpwork/llvm-project/llvm/lib/Support/Twine.cpp:65:16
    #2 0x56da3505f618 in print /fpwork/llvm-project/llvm/lib/Support/Twine.cpp:165:3
    #3 0x56da3505f618 in toVector /fpwork/llvm-project/llvm/lib/Support/Twine.cpp:34:3
    #4 0x56da3505f618 in llvm::Twine::toStringRef(llvm::SmallVectorImpl<char>&) const /fpwork/llvm-project/llvm/include/llvm/ADT/Twine.h:495:7
    #5 0x56da3505f190 in llvm::Twine::str[abi:cxx11]() const /fpwork/llvm-project/llvm/lib/Support/Twine.cpp:29:10

```